engine/swarm/secrets.md

```diff
@@ -0,0 +1,278 @@
+---
+title: Manage sensitive data (secrets) for Docker services
+description: How to securely store, retrieve, and use sensitive data used by Docker services
+keywords: swarm, secrets, credentials, sensitive strings, sensitive data, security, encryption, encryption at rest
+---
+
+## About secrets
+
+In terms of Docker Swarm services, a _secret_ is a blob of data, such as a
+password, SSH private key, SSL certificate, or another piece of data that should
+not be transmitted over a network or stored in a Dockerfile or in your
+application's source code unencrypted. In Docker 1.13 and higher, you can use
+Docker _secrets_ to centrally manage this data and securely transmit it to only
+those containers that need access to it. A _secret_ is a blob of data that is
+encrypted during transit and at rest in a Docker swarm. A given secret is only
+accessible within those containers which have been granted explicit access to it,
+and only while those containers are running.
+
+You can use secrets to manage any sensitive data which a container needs at
+runtime but you don't want to store in the image or in source control, such as:
+
+- Usernames and passwords
+- Database names
+- TLS certificates and keys
+- SSH keys
+- Generic strings or binary content (up to 500 kb in size)
+
+Another use case for using secrets is to provide a layer of abstraction between
+the container and a set of credentials. Consider a scenario where you have
+separate development, test, and production environments for your application.
+Each of these environments can have different credentials, stored in the
+development, test, and production swarms with the same secret name. Your
+containers only need to know the name of the secret in order to function in all
+three environments.
+
+## How Docker manages secrets
+
+When you add a secret to the swarm, Docker sends the secret to the swarm manager
+over a mutual TLS connection. The secret is stored in the RAFT log, which is
+encrypted. The entire RAFT log is replicated across the other managers, ensuring
+the same high availability guarantees for secrets as for the rest of the swarm
+management data.
+
+>**Warning**: RAFT data is encrypted in Docker 1.13 and higher. If any of your
+Swarm managers run an earlier version, and one of those managers becomes the
+manager of the swarm, the secrets will be stored unencrypted in that node's RAFT
+logs. Before adding any secrets, update all of your manager nodes to Docker 1.13
+to prevent secrets from being written to plain-text RAFT logs.
+
+When you grant a newly-created or running service access to a secret, the
+decrypted secret is mounted into the container in an in-memory filesystem at
+`/run/secrets/<secret_name>`. You can update a service to grant it access to
+additional secrets or revoke its access to a given secret at any time. When the
+task container stops running, the secret is unavailable to that container. When
```

Comment:
No, I was trying to cover the case where a node is running more than one task, each of which may have access to the same secret. When one of the containers stops, it doesn't have access to the secret anymore. If none of a node's task containers need access to a given secret anymore, I think the secret is flushed from the node as well, unless the node is acting as a backup master. Right? Is there a better way to put this?